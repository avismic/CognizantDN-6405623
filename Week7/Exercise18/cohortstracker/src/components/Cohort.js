// src/components/Cohort.js

export const CohortData = [
  {
    id: 1,
    cohortCode: "CHRT-1234",
    cohortName: "React Developers - Q3 2025",
    description: "A dynamic cohort focused on mastering React.js, Redux, and modern frontend workflows.",
    learners: 85,
    mode: "Online",
    status: "Ongoing"
  },
  {
    id: 2,
    cohortCode: "CHRT-5678",
    cohortName: "Full Stack Java - Q2 2025",
    description: "Comprehensive program covering Java, Spring Boot, and microservices architecture.",
    learners: 110,
    mode: "Hybrid",
    status: "Completed"
  }
];