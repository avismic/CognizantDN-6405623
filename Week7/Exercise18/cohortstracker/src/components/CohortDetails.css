/* src/components/CohortDetails.css */

.cohort-card {
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  border: 1px solid #e0e0e0;
  border-radius: 12px;
  padding: 24px;
  max-width: 400px;
  margin: 20px auto;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.08);
  background-color: #ffffff;
  position: relative;
  overflow: hidden;
  transition: transform 0.3s ease, box-shadow 0.3s ease;
}

.cohort-card:hover {
  transform: translateY(-5px);
  box-shadow: 0 8px 20px rgba(0, 0, 0, 0.12);
}

.cohort-code {
  color: #0056b3; /* A professional blue */
  margin: 0 0 8px 0;
  font-size: 1.5em;
}

.cohort-name {
  color: #333;
  margin: 0 0 16px 0;
  font-weight: 600;
}

.cohort-description {
  color: #555;
  font-size: 1em;
  line-height: 1.6;
  margin-bottom: 20px;
}

.cohort-footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-top: 16px;
  border-top: 1px solid #f0f0f0;
  color: #666;
  font-size: 0.9em;
}

.cohort-footer span {
  display: flex;
  align-items: center;
}

.status-badge {
  position: absolute;
  top: 20px;
  right: -35px;
  transform: rotate(45deg);
  padding: 5px 40px;
  font-size: 0.8em;
  font-weight: bold;
  color: white;
  box-shadow: 0 2px 5px rgba(0,0,0,0.2);
}

.status-badge.ongoing {
  background-color: #28a745; /* Green */
}

.status-badge.completed {
  background-color: #dc3545; /* Red */
}

.cohort-card-empty {
  text-align: center;
  padding: 40px;
  color: #777;
}