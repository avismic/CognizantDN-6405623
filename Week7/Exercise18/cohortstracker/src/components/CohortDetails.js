// src/components/CohortDetails.js

import React from 'react';
import './CohortDetails.css';

const CohortDetails = ({ cohort }) => {
  // ... component code
  if (!cohort) {
    return <div className="cohort-card-empty">No cohort data provided.</div>;
  }

  return (
    <div className="cohort-card">
      <div className={`status-badge ${cohort.status?.toLowerCase()}`}>{cohort.status}</div>
      <h3 className="cohort-code">{cohort.cohortCode}</h3>
      <h4 className="cohort-name">{cohort.cohortName}</h4>
      <p className="cohort-description">{cohort.description}</p>
      <div className="cohort-footer">
        <span>🎓 {cohort.learners} Learners</span>
        <span>📍 {cohort.mode}</span>
      </div>
    </div>
  );
};

// Ensure this line is exactly as follows
export default CohortDetails;