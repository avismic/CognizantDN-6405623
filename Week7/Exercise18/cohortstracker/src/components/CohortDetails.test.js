// src/components/CohortDetails.test.js

import React from 'react';
// Import render and screen from React Testing Library
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // The matchers are already included via setupTests.js

import CohortDetails from './CohortDetails';
import { CohortData } from './Cohort.js';

// The describe block remains the same
describe("Cohort Details Component", () => {

  // Test 1: should create the component
  test("should create the component", () => {
    const testCohort = CohortData[0];
    // 'render' the component into a virtual DOM
    render(<CohortDetails cohort={testCohort} />);
    // Check if an element with the cohort's name is present in the document
    const nameElement = screen.getByText(testCohort.cohortName);
    expect(nameElement).toBeInTheDocument();
  });

  // Test 2: should initialize the props (This is better tested by checking the output)
  test("should display the correct data from props", () => {
    const testCohort = CohortData[0];
    render(<CohortDetails cohort={testCohort} />);
    // We check if the content from props is rendered correctly
    expect(screen.getByText(testCohort.cohortCode)).toBeInTheDocument();
    expect(screen.getByText(testCohort.cohortName)).toBeInTheDocument();
    expect(screen.getByText(testCohort.description)).toBeInTheDocument();
  });

  // Test 3: should display cohort code in h3
  test("should display cohort code in h3", () => {
    const testCohort = CohortData[0];
    render(<CohortDetails cohort={testCohort} />);
    // Get the heading by its level (h3) and its text content
    const h3Element = screen.getByRole('heading', { level: 3, name: testCohort.cohortCode });
    expect(h3Element).toBeInTheDocument();
    // You can also check its content directly
    expect(h3Element.textContent).toBe(testCohort.cohortCode);
  });

  // Test 4: should always render same html (Snapshot Test)
  test("should always render same html", () => {
    const testCohort = CohortData[0];
    // The 'asFragment' function gives us the HTML structure
    const { asFragment } = render(<CohortDetails cohort={testCohort} />);
    // Create the snapshot
    expect(asFragment()).toMatchSnapshot();
  });

});