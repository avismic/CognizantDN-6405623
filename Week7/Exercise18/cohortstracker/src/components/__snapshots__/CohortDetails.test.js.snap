// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort Details Component should always render same html 1`] = `
<DocumentFragment>
  <div
    class="cohort-card"
  >
    <div
      class="status-badge ongoing"
    >
      Ongoing
    </div>
    <h3
      class="cohort-code"
    >
      CHRT-1234
    </h3>
    <h4
      class="cohort-name"
    >
      React Developers - Q3 2025
    </h4>
    <p
      class="cohort-description"
    >
      A dynamic cohort focused on mastering React.js, Redux, and modern frontend workflows.
    </p>
    <div
      class="cohort-footer"
    >
      <span>
        🎓 85 Learners
      </span>
      <span>
        📍 Online
      </span>
    </div>
  </div>
</DocumentFragment>
`;
